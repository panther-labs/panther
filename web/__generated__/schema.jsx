"use strict";
/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortDirEnum = exports.SeverityEnum = exports.ListRulesSortFieldsEnum = exports.ListResourcesSortFieldsEnum = exports.ListPoliciesSortFieldsEnum = exports.ListAlertsSortFieldsEnum = exports.DestinationTypeEnum = exports.ComplianceStatusEnum = exports.AnalysisTypeEnum = exports.AlertTypesEnum = exports.AlertStatusesEnum = exports.AccountTypeEnum = void 0;
var AccountTypeEnum;
(function (AccountTypeEnum) {
    AccountTypeEnum["Aws"] = "aws";
})(AccountTypeEnum = exports.AccountTypeEnum || (exports.AccountTypeEnum = {}));
var AlertStatusesEnum;
(function (AlertStatusesEnum) {
    AlertStatusesEnum["Open"] = "OPEN";
    AlertStatusesEnum["Triaged"] = "TRIAGED";
    AlertStatusesEnum["Closed"] = "CLOSED";
    AlertStatusesEnum["Resolved"] = "RESOLVED";
})(AlertStatusesEnum = exports.AlertStatusesEnum || (exports.AlertStatusesEnum = {}));
var AlertTypesEnum;
(function (AlertTypesEnum) {
    AlertTypesEnum["Rule"] = "RULE";
    AlertTypesEnum["RuleError"] = "RULE_ERROR";
})(AlertTypesEnum = exports.AlertTypesEnum || (exports.AlertTypesEnum = {}));
var AnalysisTypeEnum;
(function (AnalysisTypeEnum) {
    AnalysisTypeEnum["Rule"] = "RULE";
    AnalysisTypeEnum["Policy"] = "POLICY";
})(AnalysisTypeEnum = exports.AnalysisTypeEnum || (exports.AnalysisTypeEnum = {}));
var ComplianceStatusEnum;
(function (ComplianceStatusEnum) {
    ComplianceStatusEnum["Error"] = "ERROR";
    ComplianceStatusEnum["Fail"] = "FAIL";
    ComplianceStatusEnum["Pass"] = "PASS";
})(ComplianceStatusEnum = exports.ComplianceStatusEnum || (exports.ComplianceStatusEnum = {}));
var DestinationTypeEnum;
(function (DestinationTypeEnum) {
    DestinationTypeEnum["Slack"] = "slack";
    DestinationTypeEnum["Pagerduty"] = "pagerduty";
    DestinationTypeEnum["Github"] = "github";
    DestinationTypeEnum["Jira"] = "jira";
    DestinationTypeEnum["Opsgenie"] = "opsgenie";
    DestinationTypeEnum["Msteams"] = "msteams";
    DestinationTypeEnum["Sns"] = "sns";
    DestinationTypeEnum["Sqs"] = "sqs";
    DestinationTypeEnum["Asana"] = "asana";
    DestinationTypeEnum["Customwebhook"] = "customwebhook";
})(DestinationTypeEnum = exports.DestinationTypeEnum || (exports.DestinationTypeEnum = {}));
var ListAlertsSortFieldsEnum;
(function (ListAlertsSortFieldsEnum) {
    ListAlertsSortFieldsEnum["CreatedAt"] = "createdAt";
})(ListAlertsSortFieldsEnum = exports.ListAlertsSortFieldsEnum || (exports.ListAlertsSortFieldsEnum = {}));
var ListPoliciesSortFieldsEnum;
(function (ListPoliciesSortFieldsEnum) {
    ListPoliciesSortFieldsEnum["ComplianceStatus"] = "complianceStatus";
    ListPoliciesSortFieldsEnum["Enabled"] = "enabled";
    ListPoliciesSortFieldsEnum["Id"] = "id";
    ListPoliciesSortFieldsEnum["LastModified"] = "lastModified";
    ListPoliciesSortFieldsEnum["Severity"] = "severity";
    ListPoliciesSortFieldsEnum["ResourceTypes"] = "resourceTypes";
})(ListPoliciesSortFieldsEnum = exports.ListPoliciesSortFieldsEnum || (exports.ListPoliciesSortFieldsEnum = {}));
var ListResourcesSortFieldsEnum;
(function (ListResourcesSortFieldsEnum) {
    ListResourcesSortFieldsEnum["ComplianceStatus"] = "complianceStatus";
    ListResourcesSortFieldsEnum["Id"] = "id";
    ListResourcesSortFieldsEnum["LastModified"] = "lastModified";
    ListResourcesSortFieldsEnum["Type"] = "type";
})(ListResourcesSortFieldsEnum = exports.ListResourcesSortFieldsEnum || (exports.ListResourcesSortFieldsEnum = {}));
var ListRulesSortFieldsEnum;
(function (ListRulesSortFieldsEnum) {
    ListRulesSortFieldsEnum["Enabled"] = "enabled";
    ListRulesSortFieldsEnum["Id"] = "id";
    ListRulesSortFieldsEnum["LastModified"] = "lastModified";
    ListRulesSortFieldsEnum["LogTypes"] = "logTypes";
    ListRulesSortFieldsEnum["Severity"] = "severity";
})(ListRulesSortFieldsEnum = exports.ListRulesSortFieldsEnum || (exports.ListRulesSortFieldsEnum = {}));
var SeverityEnum;
(function (SeverityEnum) {
    SeverityEnum["Info"] = "INFO";
    SeverityEnum["Low"] = "LOW";
    SeverityEnum["Medium"] = "MEDIUM";
    SeverityEnum["High"] = "HIGH";
    SeverityEnum["Critical"] = "CRITICAL";
})(SeverityEnum = exports.SeverityEnum || (exports.SeverityEnum = {}));
var SortDirEnum;
(function (SortDirEnum) {
    SortDirEnum["Ascending"] = "ascending";
    SortDirEnum["Descending"] = "descending";
})(SortDirEnum = exports.SortDirEnum || (exports.SortDirEnum = {}));
