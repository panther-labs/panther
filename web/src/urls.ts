/**
 * Panther is a scalable, powerful, cloud-native SIEM written in Golang/React.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import { AlertDetails, PolicyDetails, ResourceDetails, RuleDetails } from 'Generated/schema';

// Typical URL encoding, allowing colons (:) to be present in the URL. Colons are safe.
// https://stackoverflow.com/questions/14872629/uriencode-and-colon
const urlEncode = (str: string) => encodeURIComponent(str).replace(/%3A/g, unescape);

const urls = {
  compliance: {
    home: () => '/cloud-security/',
    overview: () => `${urls.compliance.home()}overview/`,
    policies: {
      list: () => `${urls.compliance.home()}policies/`,
      create: () => `${urls.compliance.policies.list()}new/`,
      details: (id: PolicyDetails['id']) => `${urls.compliance.policies.list()}${urlEncode(id)}/`,
      edit: (id: PolicyDetails['id']) => `${urls.compliance.policies.details(id)}edit/`,
    },
    resources: {
      list: () => `${urls.compliance.home()}resources/`,
      details: (id: ResourceDetails['id']) => `${urls.compliance.resources.list()}${urlEncode(id)}/`, // prettier-ignore
      edit: (id: ResourceDetails['id']) => `${urls.compliance.resources.details(id)}edit/`,
    },
    sources: {
      list: () => `${urls.compliance.home()}sources/`,
      create: () => `${urls.compliance.sources.list()}new/`,
    },
  },
  logAnalysis: {
    home: () => '/log-analysis/',
    overview: () => `${urls.logAnalysis.home()}overview/`,
    rules: {
      list: () => `${urls.logAnalysis.home()}rules/`,
      create: () => `${urls.logAnalysis.rules.list()}new/`,
      details: (id: RuleDetails['id']) => `${urls.logAnalysis.rules.list()}${urlEncode(id)}/`,
      edit: (id: RuleDetails['id']) => `${urls.logAnalysis.rules.details(id)}edit/`,
    },
    alerts: {
      list: () => `${urls.logAnalysis.home()}alerts/`,
      details: (id: AlertDetails['alertId']) => `${urls.logAnalysis.alerts.list()}${urlEncode(id)}/` // prettier-ignore
    },
    sources: {
      list: () => `${urls.logAnalysis.home()}sources/`,
      create: () => `${urls.logAnalysis.sources.list()}new/`,
    },
  },
  settings: {
    overview: () => `/settings/`,
    general: () => `${urls.settings.overview()}general`,
    users: () => `${urls.settings.overview()}users`,
    destinations: () => `${urls.settings.overview()}destinations`,
  },
  account: {
    auth: {
      signIn: () => `/sign-in/`,
      forgotPassword: () => `/password-forgot/`,
      resetPassword: () => `/password-reset/`,
    },
  },
};

export default urls;
