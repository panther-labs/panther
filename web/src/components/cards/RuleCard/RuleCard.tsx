/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import GenericItemCard from 'Components/GenericItemCard';
import { Flex, Link, SimpleGrid } from 'pouncejs';
import { Link as RRLink } from 'react-router-dom';
import SeverityBadge from 'Components/badges/SeverityBadge';
import StatusBadge from 'Components/badges/StatusBadge';
import BulletedLogTypeList from 'Components/BulletedLogTypeList';
import urls from 'Source/urls';
import { RuleSummary, ComplianceStatusEnum } from 'Generated/schema';
import { formatDatetime } from 'Helpers/utils';
import useDetectionDestinations from 'Hooks/useDetectionDestinations';
import RelatedDestinations from 'Components/RelatedDestinations';
import RuleCardOptions from './RuleCardOptions';

interface RuleCardProps {
  rule: RuleSummary;
}

const RuleCard: React.FC<RuleCardProps> = ({ rule }) => {
  const {
    detectionDestinations,
    loading: loadingDetectionDestinations,
  } = useDetectionDestinations({ rule });
  return (
    <GenericItemCard>
      <GenericItemCard.Options>
        <GenericItemCard.Date date={formatDatetime(rule.lastModified)} />
        <RuleCardOptions rule={rule} />
      </GenericItemCard.Options>
      <GenericItemCard.Body>
        <Flex align="center">
          <GenericItemCard.Heading>
            <Link
              as={RRLink}
              aria-label="Link to Rule"
              to={urls.logAnalysis.rules.details(rule.id)}
            >
              {rule.displayName || rule.id}
            </Link>
          </GenericItemCard.Heading>
        </Flex>

        <SimpleGrid gap={2} columns={2}>
          <GenericItemCard.ValuesGroup>
            <GenericItemCard.Value
              label="Log Types"
              value={<BulletedLogTypeList logTypes={rule.logTypes} limit={2} />}
            />
            <GenericItemCard.Value
              label="Destinations"
              value={
                <RelatedDestinations
                  destinations={detectionDestinations}
                  loading={loadingDetectionDestinations}
                />
              }
            />
          </GenericItemCard.ValuesGroup>
          <GenericItemCard.ValuesGroup>
            <Flex ml="auto" mr={0} align="flex-end" spacing={4}>
              <StatusBadge
                status={rule.enabled ? 'ENABLED' : ComplianceStatusEnum.Error}
                disabled={!rule.enabled}
              />
              <SeverityBadge severity={rule.severity} />
            </Flex>
          </GenericItemCard.ValuesGroup>
        </SimpleGrid>
      </GenericItemCard.Body>
    </GenericItemCard>
  );
};

export default React.memo(RuleCard);
