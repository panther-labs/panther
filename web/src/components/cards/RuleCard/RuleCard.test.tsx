/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { buildRuleSummary, render } from 'test-utils';
import { SeverityEnum } from 'Generated/schema';
import urls from 'Source/urls';
import RuleCard from './index';

describe('RuleCard', () => {
  it('displays the correct Alert data in the card', async () => {
    const ruleData = buildRuleSummary();

    const { getByText } = render(<RuleCard rule={ruleData} />);

    expect(getByText(ruleData.displayName)).toBeInTheDocument();
    expect(getByText('Destinations')).toBeInTheDocument();
    expect(getByText(SeverityEnum.Info)).toBeInTheDocument();
    expect(getByText('DISABLED')).toBeInTheDocument();
  });

  it('should check links are valid', async () => {
    const ruleData = buildRuleSummary();

    const { getByAriaLabel } = render(<RuleCard rule={ruleData} />);
    expect(getByAriaLabel('Link to Rule')).toHaveAttribute(
      'href',
      urls.logAnalysis.rules.details(ruleData.id)
    );
  });
});
