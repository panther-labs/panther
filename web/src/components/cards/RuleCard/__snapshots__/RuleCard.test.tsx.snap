// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleCard renders 1`] = `
.panther-28 {
  background-color: #1d2a3f;
  border-radius: 4px;
  padding: 20px;
}

.panther-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100%;
}

.panther-26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.panther-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-1 {
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #7fd3ff;
  cursor: pointer;
}

.panther-1:hover {
  color: #ace2ff;
}

.panther-1:focus {
  color: #ace2ff;
}

.panther-1:active,
.panther-1[data-active=true] {
  color: #11a5f3;
}

.panther-0 {
  font-weight: 500;
}

.panther-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-left: auto;
  margin-right: 0;
}

.panther-3 {
  outline: none;
  line-height: 1;
  height: 32px;
  width: 32px;
  -webkit-transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 99999px;
  border: 1px solid;
  border-color: transparent;
  background-color: transparent;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-3:hover {
  background-color: rgba(0,129,197,0.3);
}

.panther-3:focus {
  background-color: rgba(0,129,197,0.3);
}

.panther-3:active,
.panther-3[data-active=true] {
  background-color: #0081c5;
}

.panther-2 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  color: currentColor;
}

.panther-25 {
  display: grid;
  grid-gap: 8px;
  grid-template-columns: repeat(2, 1fr);
}

.panther-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-13>*:not(:last-child) {
  margin-right: 32px;
}

.panther-12 {
  margin-top: 16px;
}

.panther-6 {
  color: #bdbdbd;
  font-size: 0.563rem;
  margin-bottom: 1px;
  font-weight: 500;
}

.panther-11 {
  font-size: 0.875rem;
  opacity: 1;
}

.panther-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 4px;
}

.panther-10>*:not(:last-child) {
  margin-right: 24px;
}

.panther-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-9>*:not(:last-child) {
  margin-right: 8px;
}

.panther-7 {
  width: 12px;
  height: 12px;
  background-color: hsl(288, 38%, 68%);
  border-radius: 99999px;
}

.panther-8 {
  font-size: 0.75rem;
}

.panther-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-left: auto;
  margin-right: 0;
}

.panther-23>*:not(:last-child) {
  margin-right: 8px;
}

.panther-21 {
  opacity: 0.5;
}

.panther-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #4f4f4f;
}

.panther-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid;
  border-color: #bdbdbd;
  background-color: transparent;
}

<div>
  <section
    class="panther-28"
  >
    <div
      class="panther-27"
    >
      <div
        class="panther-26"
      >
        <div
          class="panther-5"
        >
          <a
            aria-label="Link to Rule"
            class="panther-1"
            href="/log-analysis/rules/4ce135b7-005f-4a98-8a69-9b9d3b372bdb/"
          >
            <h4
              class="panther-0"
            >
              array
            </h4>
          </a>
          <div
            class="panther-4"
          >
            <button
              aria-controls="menu--1"
              aria-haspopup="true"
              aria-label="Rule Options"
              aria-pressed="false"
              class="panther-3"
              data-reach-menu-button=""
              id="menu-button--menu--1"
              type="button"
            >
              <svg
                class="panther-2"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 10C4.9 10 4 10.9 4 12C4 13.1 4.9 14 6 14C7.1 14 8 13.1 8 12C8 10.9 7.1 10 6 10ZM18 10C16.9 10 16 10.9 16 12C16 13.1 16.9 14 18 14C19.1 14 20 13.1 20 12C20 10.9 19.1 10 18 10ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10Z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="panther-25"
        >
          <div
            class="panther-13"
          >
            <dl
              class="panther-12"
            >
              <dt
                aria-labelledby="log-typesobject-object"
                class="panther-6"
              >
                Log Types
              </dt>
              <dd
                aria-labelledby="log-typesobject-object"
                class="panther-11"
              >
                <div
                  class="panther-10"
                >
                  <div
                    class="panther-9"
                  >
                    <span
                      class="panther-7"
                    />
                    <span
                      class="panther-8"
                    >
                      AI
                    </span>
                  </div>
                </div>
              </dd>
            </dl>
          </div>
          <div
            class="panther-13"
          >
            <div
              class="panther-23"
            >
              <dl
                class="panther-12"
              >
                <dt
                  aria-labelledby="threshold550"
                  class="panther-6"
                >
                  Threshold
                </dt>
                <dd
                  aria-labelledby="threshold550"
                  class="panther-11"
                >
                  550
                </dd>
              </dl>
              <dl
                class="panther-12"
              >
                <dt
                  aria-labelledby="time-created2020-09-24-1209-gmt3"
                  class="panther-6"
                >
                  Time Created
                </dt>
                <dd
                  aria-labelledby="time-created2020-09-24-1209-gmt3"
                  class="panther-11"
                >
                  2020-09-24 12:09 GMT+3
                </dd>
              </dl>
              <div
                class="panther-21"
              >
                <div
                  aria-atomic="true"
                  class="panther-20"
                  role="status"
                >
                  DISABLED
                </div>
              </div>
              <div
                aria-atomic="true"
                class="panther-22"
                role="status"
              >
                INFO
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`RuleCard renders an expanded dropdown 1`] = `
.panther-28 {
  background-color: #1d2a3f;
  border-radius: 4px;
  padding: 20px;
}

.panther-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100%;
}

.panther-26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.panther-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-1 {
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #7fd3ff;
  cursor: pointer;
}

.panther-1:hover {
  color: #ace2ff;
}

.panther-1:focus {
  color: #ace2ff;
}

.panther-1:active,
.panther-1[data-active=true] {
  color: #11a5f3;
}

.panther-0 {
  font-weight: 500;
}

.panther-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-left: auto;
  margin-right: 0;
}

.panther-3 {
  outline: none;
  line-height: 1;
  height: 32px;
  width: 32px;
  -webkit-transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 99999px;
  border: 1px solid;
  border-color: transparent;
  background-color: transparent;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-3:hover {
  background-color: rgba(0,129,197,0.3);
}

.panther-3:focus {
  background-color: rgba(0,129,197,0.3);
}

.panther-3:active,
.panther-3[data-active=true] {
  background-color: #0081c5;
}

.panther-2 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  color: currentColor;
}

.panther-25 {
  display: grid;
  grid-gap: 8px;
  grid-template-columns: repeat(2, 1fr);
}

.panther-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-13>*:not(:last-child) {
  margin-right: 32px;
}

.panther-12 {
  margin-top: 16px;
}

.panther-6 {
  color: #bdbdbd;
  font-size: 0.563rem;
  margin-bottom: 1px;
  font-weight: 500;
}

.panther-11 {
  font-size: 0.875rem;
  opacity: 1;
}

.panther-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 4px;
}

.panther-10>*:not(:last-child) {
  margin-right: 24px;
}

.panther-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-9>*:not(:last-child) {
  margin-right: 8px;
}

.panther-7 {
  width: 12px;
  height: 12px;
  background-color: hsl(288, 38%, 68%);
  border-radius: 99999px;
}

.panther-8 {
  font-size: 0.75rem;
}

.panther-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-left: auto;
  margin-right: 0;
}

.panther-23>*:not(:last-child) {
  margin-right: 8px;
}

.panther-21 {
  opacity: 0.5;
}

.panther-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #4f4f4f;
}

.panther-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid;
  border-color: #bdbdbd;
  background-color: transparent;
}

<div>
  <section
    class="panther-28"
  >
    <div
      class="panther-27"
    >
      <div
        class="panther-26"
      >
        <div
          class="panther-5"
        >
          <a
            aria-label="Link to Rule"
            class="panther-1"
            href="/log-analysis/rules/4ce135b7-005f-4a98-8a69-9b9d3b372bdb/"
          >
            <h4
              class="panther-0"
            >
              array
            </h4>
          </a>
          <div
            class="panther-4"
          >
            <button
              aria-controls="menu--4"
              aria-haspopup="true"
              aria-label="Rule Options"
              aria-pressed="false"
              class="panther-3"
              data-reach-menu-button=""
              id="menu-button--menu--4"
              type="button"
            >
              <svg
                class="panther-2"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 10C4.9 10 4 10.9 4 12C4 13.1 4.9 14 6 14C7.1 14 8 13.1 8 12C8 10.9 7.1 10 6 10ZM18 10C16.9 10 16 10.9 16 12C16 13.1 16.9 14 18 14C19.1 14 20 13.1 20 12C20 10.9 19.1 10 18 10ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10Z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="panther-25"
        >
          <div
            class="panther-13"
          >
            <dl
              class="panther-12"
            >
              <dt
                aria-labelledby="log-typesobject-object"
                class="panther-6"
              >
                Log Types
              </dt>
              <dd
                aria-labelledby="log-typesobject-object"
                class="panther-11"
              >
                <div
                  class="panther-10"
                >
                  <div
                    class="panther-9"
                  >
                    <span
                      class="panther-7"
                    />
                    <span
                      class="panther-8"
                    >
                      AI
                    </span>
                  </div>
                </div>
              </dd>
            </dl>
          </div>
          <div
            class="panther-13"
          >
            <div
              class="panther-23"
            >
              <dl
                class="panther-12"
              >
                <dt
                  aria-labelledby="threshold550"
                  class="panther-6"
                >
                  Threshold
                </dt>
                <dd
                  aria-labelledby="threshold550"
                  class="panther-11"
                >
                  550
                </dd>
              </dl>
              <dl
                class="panther-12"
              >
                <dt
                  aria-labelledby="time-created2020-09-24-1209-gmt3"
                  class="panther-6"
                >
                  Time Created
                </dt>
                <dd
                  aria-labelledby="time-created2020-09-24-1209-gmt3"
                  class="panther-11"
                >
                  2020-09-24 12:09 GMT+3
                </dd>
              </dl>
              <div
                class="panther-21"
              >
                <div
                  aria-atomic="true"
                  class="panther-20"
                  role="status"
                >
                  DISABLED
                </div>
              </div>
              <div
                aria-atomic="true"
                class="panther-22"
                role="status"
              >
                INFO
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
